src/framework/entity.js

```diff
@@ -30,6 +30,33 @@ import { getApplication } from './globals.js';
  * @import { SpriteComponent } from './components/sprite/component.js'
  */
 
+/**
+ * @typedef {{
+ *     anim: AnimComponent;
+ *     animation: AnimationComponent;
+ *     audiolistener: AudioListenerComponent;
+ *     button: ButtonComponent;
+ *     camera: CameraComponent;
+ *     collision: CollisionComponent;
+ *     element: ElementComponent;
+ *     gsplat: GSplatComponent;
+ *     layoutchild: LayoutChildComponent;
+ *     layoutgroup: LayoutGroupComponent;
+ *     light: LightComponent;
+ *     model: ModelComponent;
+ *     particlesystem: ParticleSystemComponent;
+ *     render: RenderComponent;
+ *     rigidbody: RigidBodyComponent;
+ *     screen: ScreenComponent;
+ *     script: ScriptComponent;
+ *     scrollbar: ScrollbarComponent;
+ *     scrollview: ScrollViewComponent;
+ *     sound: SoundComponent;
+ *     sprite: SpriteComponent;
+ * }} ComponentMap A map of component names to their class types. Facilitates typings for
+ * {@link Entity#addComponent}.
```

Comment:
Not that I ever used `ZoneComponent`, but shouldn't it be in here? And `AudioSourceComponentSystem`, or rather not because it's deprecated?

Anyway, happy to see some stricter types!

Reply:
Good point, that explains why `JointComponent` isn't here either I guess. First I hoped there could be a single-source-of-truth like:

```js
/**
 * @typedef {{
 *    [K in keyof ComponentSystemRegistry]:
 *        ComponentSystemRegistry[K] extends ComponentSystem ? K : never
 * }} ComponentTypeProps
 * @typedef {ComponentTypeProps[keyof ComponentTypeProps]} ComponentSystemNames
 * @typedef {{[K in ComponentSystemNames]: InstanceType<ComponentSystemRegistry[K]['ComponentType']>}} ComponentMap
 */
```
(pretty much using the fact that the `ComponentSystemRegistry` class is referencing everything that's needed)

Filtering ignored components afterwards feels like overkill though, so your approach is the most straight-forward one.